libcxx/include/version

```diff
@@ -510,7 +510,9 @@ __cpp_lib_void_t                                        201411L <type_traits>
 # undef  __cpp_lib_bind_front
 # define __cpp_lib_bind_front                           202306L
 # define __cpp_lib_bitset                               202306L
-// # define __cpp_lib_constexpr_new                        202406L
+# if !defined(_LIBCPP_ABI_VCRUNTIME)
```

Comment:
Do we want a check for `__cpp_constexpr >= 202406L` here too? `constexpr operator new` has been valid since C++23 but isn't usable without compiler support, and users might check `defined(__cpp_lib_constexpr_new)` instead of `defined(__cpp_lib_constexpr_new) && __cpp_constexpr >= 202406L`

Reply:
> Do we want a check for `__cpp_constexpr >= 202406L` here too? `constexpr operator new` has been valid since C++23 but isn't usable without compiler support,

I'm pretty sure that `constexpr operator new` in this paper doesn't require any constexpr feature after C++11. So the library changes don't require any chang in the compiler.

> and users might check `defined(__cpp_lib_constexpr_new)` instead of `defined(__cpp_lib_constexpr_new) && __cpp_constexpr >= 202406L`

I don't think it's meaningful to check `__cpp_lib_constexpr_new`. The approach for the core pary in #104586 doesn't require library changes.

In other words, under Clang's current (WIP) approach for constexpr placement new, the core and library changes are independent.

Reply:
Your right, I was thinking that `operator new` could never be called in a constant expression, which might have been ill-formed NDR before C++23, but I forgot it could just be called directly `operator new(1zu, p)` without a new-expression. Which I guess is the only thing this patch adds support for with clang.

Looks like the GCC implementation *does* check if `operator new` is marked `constexpr`: https://godbolt.org/z/hYaYYvaKc so this is needed for GCC at least.